building nix-du fails with "error: 'requireGcStore' was not declared in this scope"

### Describe the bug
running `nix profile upgrade '.*'` tries to rebuild nix-du 0.5.0 (not sure why) and fails with
```
error: builder for '/nix/store/xgmya4il4zzbbfc38zwdpcjbz6ksqhfm-nix-du-0.5.0.drv' failed with exit code 101;
       last 10 log lines:
       >   cargo:warning=   21 | #define findroots(store) requireGcStore(*store).findRoots(false)
       >   cargo:warning=      |                          ^~~~~~~~~~~~~~
       >   exit status: 1
       >
       >   --- stderr
       >
       >
       >   error occurred: Command "/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-Wall" "-Wextra" "-std=c++17" "-DNIXVER=207" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/nix-du-a6e05f9c2d696b71/out/wrapper.o" "-c" "wrapper.cpp" with args "c++" did not execute successfully (status code exit status: 1).
       >
       >
       For full logs, run 'nix log /nix/store/xgmya4il4zzbbfc38zwdpcjbz6ksqhfm-nix-du-0.5.0.drv'.
```

### Steps To Reproduce
Steps to reproduce the behavior:
i'm not sure but these are true for me:
1. have nix-du installed (`33 flake:nixpkgs#legacyPackages.x86_64-linux.nix-du github:NixOS/nixpkgs/b06d35b406c2396dd1611dc8601138fa3d06ee60#legacyPackages.x86_64-linux.nix-du /nix/store/g3p9wk337bj4m7c0blq6r2nxgrz18nyc-nix-du-0.5.0`)
2. `nix profile upgrade '.*'`

### Expected behavior
nix-du builds successfully

### Additional context
<details>
<summary>nix log /nix/store/xgmya4il4zzbbfc38zwdpcjbz6ksqhfm-nix-du-0.5.0.drv</summary>
<pre>
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/vzwbsy04xlsc451hf8zzcrwkqjsdjkwx-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/km5fszgmfy743s08hw48frsrw9plkldf-nix-du-0.5.0-vendor.tar.gz
Finished cargoSetupPostUnpackHook
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing cargoSetupPostPatchHook
Validating consistency between /build/source//Cargo.lock and /build/nix-du-0.5.0-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing cargoBuildHook
++ env CC_x86_64-unknown-linux-gnu=/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++ cargo build -j 4 --target x86_64-unknown-linux-gnu --frozen --release
[0m[0m[1m[32m   Compiling[0m libc v0.2.106
[0m[0m[1m[32m   Compiling[0m cfg-if v1.0.0
[0m[0m[1m[32m   Compiling[0m proc-macro2 v1.0.32
[0m[0m[1m[32m   Compiling[0m unicode-xid v0.2.2
[0m[0m[1m[32m   Compiling[0m memchr v2.4.1
[0m[0m[1m[32m   Compiling[0m autocfg v1.0.1
[0m[0m[1m[32m   Compiling[0m syn v1.0.81
[0m[0m[1m[32m   Compiling[0m glob v0.3.0
[0m[0m[1m[32m   Compiling[0m version_check v0.9.4
[0m[0m[1m[32m   Compiling[0m serde v1.0.130
[0m[0m[1m[32m   Compiling[0m log v0.4.14
[0m[0m[1m[32m   Compiling[0m ppv-lite86 v0.2.15
[0m[0m[1m[32m   Compiling[0m regex-syntax v0.6.25
[0m[0m[1m[32m   Compiling[0m unicode-width v0.1.9
[0m[0m[1m[32m   Compiling[0m proc-macro-hack v0.5.19
[0m[0m[1m[32m   Compiling[0m minimal-lexical v0.2.1
[0m[0m[1m[32m   Compiling[0m siphasher v0.3.7
[0m[0m[1m[32m   Compiling[0m humantime v2.1.0
[0m[0m[1m[32m   Compiling[0m ansi_term v0.11.0
[0m[0m[1m[32m   Compiling[0m bitflags v1.3.2
[0m[0m[1m[32m   Compiling[0m either v1.6.1
[0m[0m[1m[32m   Compiling[0m termcolor v1.1.2
[0m[0m[1m[32m   Compiling[0m bindgen v0.59.2
[0m[0m[1m[32m   Compiling[0m vec_map v0.8.2
[0m[0m[1m[32m   Compiling[0m strsim v0.8.0
[0m[0m[1m[32m   Compiling[0m peeking_take_while v0.1.2
[0m[0m[1m[32m   Compiling[0m shlex v1.1.0
[0m[0m[1m[32m   Compiling[0m lazy_static v1.4.0
[0m[0m[1m[32m   Compiling[0m lazycell v1.3.0
[0m[0m[1m[32m   Compiling[0m rustc-hash v1.1.0
[0m[0m[1m[32m   Compiling[0m pkg-config v0.3.22
[0m[0m[1m[32m   Compiling[0m hashbrown v0.11.2
[0m[0m[1m[32m   Compiling[0m palette v0.6.0
[0m[0m[1m[32m   Compiling[0m cc v1.0.71
[0m[0m[1m[32m   Compiling[0m fixedbitset v0.4.0
[0m[0m[1m[32m   Compiling[0m same-file v1.0.6
[0m[0m[1m[32m   Compiling[0m human-size v0.4.1
[0m[0m[1m[32m   Compiling[0m humansize v1.1.1
[0m[0m[1m[32m   Compiling[0m libloading v0.7.3
[0m[0m[1m[32m   Compiling[0m num-traits v0.2.14
[0m[0m[1m[32m   Compiling[0m indexmap v1.7.0
[0m[0m[1m[32m   Compiling[0m nom v7.1.0
[0m[0m[1m[32m   Compiling[0m clang-sys v1.3.1
[0m[0m[1m[32m   Compiling[0m phf_shared v0.9.0
[0m[0m[1m[32m   Compiling[0m walkdir v2.3.2
[0m[0m[1m[32m   Compiling[0m aho-corasick v0.7.18
[0m[0m[1m[32m   Compiling[0m getrandom v0.2.3
[0m[0m[1m[32m   Compiling[0m atty v0.2.14
[0m[0m[1m[32m   Compiling[0m term_size v0.3.2
[0m[0m[1m[32m   Compiling[0m which v4.2.4
[0m[0m[1m[32m   Compiling[0m quote v1.0.10
[0m[0m[1m[32m   Compiling[0m rand_core v0.6.3
[0m[0m[1m[32m   Compiling[0m textwrap v0.11.0
[0m[0m[1m[32m   Compiling[0m regex v1.5.4
[0m[0m[1m[32m   Compiling[0m petgraph v0.6.0
[0m[0m[1m[32m   Compiling[0m approx v0.5.0
[0m[0m[1m[32m   Compiling[0m rand_chacha v0.3.1
[0m[0m[1m[32m   Compiling[0m clap v2.33.3
[0m[0m[1m[32m   Compiling[0m toml v0.5.8
[0m[0m[1m[32m   Compiling[0m env_logger v0.9.0
[0m[0m[1m[32m   Compiling[0m cexpr v0.6.0
[0m[0m[1m[32m   Compiling[0m rand v0.8.4
[0m[0m[1m[32m   Compiling[0m find-crate v0.6.3
[0m[0m[1m[32m   Compiling[0m phf_generator v0.9.1
[0m[0m[1m[32m   Compiling[0m enum-map-derive v0.6.0
[0m[0m[1m[32m   Compiling[0m palette_derive v0.6.0
[0m[0m[1m[32m   Compiling[0m phf_macros v0.9.0
[0m[0m[1m[32m   Compiling[0m enum-map v1.1.1
[0m[0m[1m[32m   Compiling[0m phf v0.9.0
[0m[0m[1m[32m   Compiling[0m nix-du v0.5.0 (/build/source)
The following warnings were emitted during compilation:

[0m[0m[1m[33mwarning[0m[1m:[0m wrapper.cpp: In lambda function:
[0m[0m[1m[33mwarning[0m[1m:[0m wrapper.cpp:21:26: error: 'requireGcStore' was not declared in this scope
[0m[0m[1m[33mwarning[0m[1m:[0m    21 | #define findroots(store) requireGcStore(*store).findRoots(false)
[0m[0m[1m[33mwarning[0m[1m:[0m       |                          ^~~~~~~~~~~~~~
[0m[0m[1m[33mwarning[0m[1m:[0m wrapper.cpp:21:26: note: in definition of macro 'findroots'
[0m[0m[1m[33mwarning[0m[1m:[0m    21 | #define findroots(store) requireGcStore(*store).findRoots(false)
[0m[0m[1m[33mwarning[0m[1m:[0m       |                          ^~~~~~~~~~~~~~

[0m[0m[1m[31merror[0m[1m:[0m failed to run custom build command for `nix-du v0.5.0 (/build/source)`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/nix-du-e4df9bb2bc40e2d3/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=wrapper.hpp
  cargo:rerun-if-changed=wrapper.cpp
  cargo:rerun-if-env-changed=NIX_MAIN_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=NIX_MAIN_STATIC
  cargo:rerun-if-env-changed=NIX_MAIN_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=SYSROOT
  cargo:rerun-if-env-changed=NIX_MAIN_STATIC
  cargo:rerun-if-env-changed=NIX_MAIN_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rustc-link-search=native=/nix/store/9r5zb6mdxvlfg6jbjrgld3di3pb18mj9-nix-2.8.0/lib
  cargo:rustc-link-lib=nixmain
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=NIX_MAIN_STATIC
  cargo:rerun-if-env-changed=NIX_MAIN_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  Found nix version 2.8.0
  TARGET = Some("x86_64-unknown-linux-gnu")
  HOST = Some("x86_64-unknown-linux-gnu")
  CXX_x86_64-unknown-linux-gnu = Some("/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++")
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CXXFLAGS = None
  CXXFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-Wall" "-Wextra" "-std=c++17" "-DNIXVER=207" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/nix-du-a6e05f9c2d696b71/out/wrapper.o" "-c" "wrapper.cpp"
  cargo:warning=wrapper.cpp: In lambda function:
  cargo:warning=wrapper.cpp:21:26: error: 'requireGcStore' was not declared in this scope
  cargo:warning=   21 | #define findroots(store) requireGcStore(*store).findRoots(false)
  cargo:warning=      |                          ^~~~~~~~~~~~~~
  cargo:warning=wrapper.cpp:21:26: note: in definition of macro 'findroots'
  cargo:warning=   21 | #define findroots(store) requireGcStore(*store).findRoots(false)
  cargo:warning=      |                          ^~~~~~~~~~~~~~
  exit status: 1

  --- stderr


  error occurred: Command "/nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/c++" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-Wall" "-Wextra" "-std=c++17" "-DNIXVER=207" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/nix-du-a6e05f9c2d696b71/out/wrapper.o" "-c" "wrapper.cpp" with args "c++" did not execute successfully (status code exit status: 1).
</pre>
</details>
Add any other context about the problem here.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
;  nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.20-200.fc35.x86_64, Fedora Linux, 35 (Workstation Edition), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - channels(mkhl): `"nixpkgs"`
 - nixpkgs: `/home/mkhl/.nix-defexpr/channels/nixpkgs`
```

--- a/pkgs/tools/package-management/nix-du/default.nix
+++ b/pkgs/tools/package-management/nix-du/default.nix
@@ -12,16 +12,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "nix-du";
-  version = "0.5.0";
+  version = "0.5.1";
 
   src = fetchFromGitHub {
     owner = "symphorien";
     repo = "nix-du";
     rev = "v${version}";
-    sha256 = "sha256-NnoCoiYTXci7f9MXsxEhoYHwPsaAAuK2wSrDye2+SYM=";
+    sha256 = "sha256-LOs+geYOiAigxwc4VD0FiZQjAnIrcV9ncyPuwGhS92E=";
   };
 
-  cargoSha256 = "sha256-X5181gZtvnUHZAqMjNARRVFq12LuLJfPP5ZOWxBRxXk=";
+  cargoSha256 = "sha256-aEm+SQgE63ZWpb2kXavyoiq2rVkaebFw8kqWPMr2aMA=";
 
   doCheck = true;
   checkInputs = [ nix graphviz ];

